internal/streamingnode/server/wal/interceptors/segment/manager/seal_queue.go

```diff
@@ -0,0 +1,195 @@
+	q.cond.L.Unlock()
+}
+
+// transferSegmentStateIntoSealed transfers the segment state into sealed.
+func (q *sealQueue) transferSegmentStateIntoSealed(ctx context.Context, segments ...*segmentAllocManager) ([]*segmentAllocManager, map[int64]map[string][]*segmentAllocManager) {
+	// undone sealed segment should be done at next time.
+	undone := make([]*segmentAllocManager, 0)
+	sealedSegments := make(map[int64]map[string][]*segmentAllocManager)
+	for _, segment := range segments {
+		if segment.GetState() == streamingpb.SegmentAssignmentState_SEGMENT_ASSIGNMENT_STATE_GROWING {
+			tx := segment.BeginModification()
+			tx.IntoSealed()
+			if err := tx.Commit(ctx); err != nil {
+				q.logger.Warn("seal segment failed at commit", zap.Int64("segmentID", segment.GetSegmentID()), zap.Error(err))
+				undone = append(undone, segment)
+				continue
+			}
+		}
+		// assert here.
+		if segment.GetState() != streamingpb.SegmentAssignmentState_SEGMENT_ASSIGNMENT_STATE_SEALED {
+			panic("unreachable code: segment should be sealed here")
+		}
+
+		// if there'are flying acks, wait them acked, delay the sealed at next retry.
+		ackSem := segment.AckSem()
+		if ackSem > 0 {
+			undone = append(undone, segment)
+			q.logger.Info("segment has been sealed, but there are flying acks, delay it", zap.Int64("segmentID", segment.GetSegmentID()), zap.Int32("ackSem", ackSem))
+			continue
+		}
+
+		// collect all sealed segments and no flying ack segment.
+		if _, ok := sealedSegments[segment.GetCollectionID()]; !ok {
+			sealedSegments[segment.GetCollectionID()] = make(map[string][]*segmentAllocManager)
+		}
+		if _, ok := sealedSegments[segment.GetCollectionID()][segment.GetVChannel()]; !ok {
+			sealedSegments[segment.GetCollectionID()][segment.GetVChannel()] = make([]*segmentAllocManager, 0)
+		}
+		sealedSegments[segment.GetCollectionID()][segment.GetVChannel()] = append(sealedSegments[segment.GetCollectionID()][segment.GetVChannel()], segment)
+	}
+	return undone, sealedSegments
+}
+
+// sendFlushMessageIntoWAL sends a flush message into wal.
+func (m *sealQueue) sendFlushMessageIntoWAL(ctx context.Context, collectionID int64, vchannel string, segments []*segmentAllocManager) error {
+	segmentIDs := make([]int64, 0, len(segments))
+	for _, segment := range segments {
+		segmentIDs = append(segmentIDs, segment.GetSegmentID())
+	}
+	msg, err := m.createNewFlushMessage(collectionID, vchannel, segmentIDs)
+	if err != nil {
+		return errors.Wrap(err, "at create new flush message")
+	}
+
+	msgID, err := m.wal.Get().Append(ctx, msg)
+	if err != nil {
+		m.logger.Warn("send flush message into wal failed", zap.Int64("collectionID", collectionID), zap.String("vchannel", vchannel), zap.Int64s("segmentIDs", segmentIDs), zap.Error(err))
+		return err
+	}
+	m.logger.Info("send flush message into wal", zap.Int64("collectionID", collectionID), zap.String("vchannel", vchannel), zap.Int64s("segmentIDs", segmentIDs), zap.Any("msgID", msgID))
```

Comment:
Use debug level instead Info level

Reply:
same as above, it's a segment level log.